sky/sdk/example/game/main.dart

```diff
@@ -90,13 +91,86 @@ class GameDemoApp extends App {
 
   Widget _buildMainScene(navigator, route) {
     return new Center(
-      child: new RaisedButton(
-        child: new Text("Play"),
+      child: new TextureButton(
         onPressed: () {
           _game = new GameDemoWorld(_app, navigator, _loader, _spriteSheet, _spriteSheetUI);
           navigator.pushNamed('/game');
-        }
+        },
+        texture: _spriteSheetUI["btn_play_up.png"],
+        textureDown: _spriteSheetUI["btn_play_down.png"],
+        width: 128.0,
+        height: 128.0
       )
     );
   }
 }
+
+class TextureButton extends ButtonBase {
+  TextureButton({
+    Key key,
+    this.onPressed,
+    this.texture,
+    this.textureDown,
+    this.width: 128.0,
+    this.height: 128.0
+  }) : super(key: key);
+
+  Function onPressed;
+  Texture texture;
+  Texture textureDown;
+  double width;
+  double height;
+
+  int _token = 0;
+
+  void initState() {
+    highlight = false;
+    super.initState();
+  }
+
+  void syncFields(TextureButton source) {
+    onPressed = source.onPressed;
+    texture = source.texture;
+    textureDown = source.textureDown;
+    width = source.width;
+    height = source.height;
+    _token++;
+    super.syncFields(source);
+  }
+
+  Widget buildContent() {
+    return new Listener(
+      child: new Container(
+        width: width,
+        height: height,
+        child: new CustomPaint(
+          callback: paintCallback,
+          token: _token
+        )
+      ),
+      onPointerUp: (_) {
+        if (onPressed != null) onPressed();
+      }
+    );
+  }
+
+  void setState(Function fn) {
+    _token++;
```

Comment:
if you're trying to make the token change when highlight changes, I think the way to do this is to not have a token per se, but just have the build function create a token each time it creates the CustomPaint, where that token is a class that is == to other instances of the class so long as texture, textureDown, highlight, width, and height are the same.